service/common/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -166,12 +166,15 @@ public static Collection<String> getAccessDeniedHints() {
 
   static int mapExceptionToResponseCode(RuntimeException rex) {
     // Cloud exceptions
-    if (rex instanceof S3Exception
-        || rex instanceof AzureException
-        || rex instanceof StorageException) {
+    if (isCloudException(rex)) {
       return mapCloudExceptionToResponseCode(rex);
     }
 
+    Throwable rootCause = ExceptionUtils.getRootCause(rex);
```

Comment:
I vaguely remember this coming up before, but I suspect we should inspect the entire cause chain rather than just the root cause

Reply:
Thanks for the reminder, updated!